plugin-k8s/src/cgroup_v2.rs

```diff
@@ -40,48 +40,24 @@ pub fn is_accessible_dir(path: &Path) -> bool {
     path.is_dir()
 }
 
-/// Retrieves the name of a cgroup without its prefix.
-///
-/// ## Example
-/// ```ignore
-/// use std::path::PathBuf;
-/// 
-/// let path = PathBuf::from("myPath/kubepods-burstable.slice/kubepods-burstable-podABCD");
-/// assert_eq!(retrieve_name(&path, &"kubepods-burstable-".to_string()).unwrap(),"podABCD");
-/// ```
-fn retrieve_name(path: &Path, prefix: &str) -> anyhow::Result<String> {
-    // Get the last component of the path (file or directory name)
-    if path.as_os_str().is_empty() {
-        anyhow::bail!("Path can't be empty")
-    }
-    if prefix != "" {
-        let file_name = path.file_name().unwrap().to_string_lossy();
-        let file_strip_prefix = file_name.strip_prefix(prefix).unwrap_or(&file_name);
-        let file_strip_suffix = file_strip_prefix.strip_suffix(&".slice").unwrap_or(file_strip_prefix);
-        return Ok(file_strip_suffix.to_owned());
-    }
-    Ok("".to_owned())
-}
 
 /// Returns a Vector of CgroupV2MetricFile associated to pods availables under a given directory.
-fn list_metric_file_in_dir(root_directory_path: &String, prefix: &String) -> anyhow::Result<Vec<CgroupV2MetricFile>> {
-    let root_path = Path::new(root_directory_path);
-    let prefix_path = Path::new(prefix);
-    let dir = root_path.join(prefix_path);
+fn list_metric_file_in_dir(root_directory_path: &Path) -> anyhow::Result<Vec<CgroupV2MetricFile>> {
     let mut vec_file_metric: Vec<CgroupV2MetricFile> = Vec::new();
-    let entries = fs::read_dir(dir)?;
+    let entries = fs::read_dir(root_directory_path)?;
     for entry in entries {
         let mut path = entry?.path();
         if path.is_dir() {
-            let _dir_name = path.file_name().expect("Impossible to write dir name");
-            let truncated_prefix = prefix.strip_suffix(".slice/").unwrap_or(&prefix);
-            let mut new_prefix = truncated_prefix.to_owned();
-            new_prefix.push_str("-");
-            let name = retrieve_name(&path, &new_prefix.to_owned())?;
+            let dir_name = path.file_name().expect("Impossible get dir name as String").to_os_string().into_string().map_err(|e| anyhow::Error::msg(format!("Conversion from Path to String error: {:?}", e)))?;
```

Comment:
A suggestion that can be applied in multiple places, to make the code more readable:
```suggestion
            let dir_name = path.file_name().unwrap().to_str().ok_or(anyhow!("filename is not valid UTF-8: {}", path.display())?;
```
A simpler way to express the conversion of a path to a UTF-8 string would be `.to_str().ok_or(error)`.

Regarding the `file_name()`, according to the doc it only returns `None` when the path ends with `..`, which cannot happen here because you are listing the files of a directory. Hence I think `unwrap()` can be appropriate.